ksqldb-engine/src/test/java/io/confluent/ksql/physical/pull/operators/KeyedTableLookupOperatorTest.java

```diff
@@ -142,4 +151,51 @@ public void shouldLookupRowsForMultipleKeys() {
     assertThat(lookupOperator.next(), is(nullValue()));
     assertThat(lookupOperator.getReturnedRowCount(), is(3L));
   }
+
+  @Test
+  public void shouldLookupRowsForRangeKeySinglePartition() {
+    //Given:
+    when(KEY3.getOperator()).thenReturn(ConstraintOperator.LESS_THAN_OR_EQUAL);
+    final List<KsqlPartitionLocation> singleKeyPartitionLocations = new ArrayList<>();
+    singleKeyPartitionLocations.add(new KsLocator.PartitionLocation(
+      Optional.of(ImmutableSet.of(KEY1, KEY3)), 1, ImmutableList.of(node1)));
+
+    final KeyedTableLookupOperator lookupOperator = new KeyedTableLookupOperator(materialization, logicalNode);
+    when(materialization.nonWindowed()).thenReturn(nonWindowedTable);
+    when(materialization.nonWindowed().get(1,null, GKEY3)).thenReturn(Arrays.asList(ROW1, ROW3).iterator());
+
+    lookupOperator.setPartitionLocations(singleKeyPartitionLocations);
+    lookupOperator.open();
+
+    //Then:
+    assertThat(lookupOperator.next(), is(ROW1));
+    assertThat(lookupOperator.next(), is(ROW3));
+    assertThat(lookupOperator.getReturnedRowCount(), is(2L));
+  }
+
+  @Test
+  public void shouldLookupRowsForRangeKeyMultiplePartitions() {
+    //Given:
+    when(KEY2.getOperator()).thenReturn(ConstraintOperator.LESS_THAN_OR_EQUAL);
+    when(KEY3.getOperator()).thenReturn(ConstraintOperator.LESS_THAN_OR_EQUAL);
+    final List<KsqlPartitionLocation> multipleKeysPartitionLocations = new ArrayList<>();
+    multipleKeysPartitionLocations.add(new KsLocator.PartitionLocation(
+      Optional.of(ImmutableSet.of(KEY1, KEY2)), 1, ImmutableList.of(node1)));
+    multipleKeysPartitionLocations.add(new KsLocator.PartitionLocation(
+      Optional.of(ImmutableSet.of(KEY3, KEY4)), 3, ImmutableList.of(node3)));
+
+    final KeyedTableLookupOperator lookupOperator = new KeyedTableLookupOperator(materialization, logicalNode);
+    when(materialization.nonWindowed()).thenReturn(nonWindowedTable);
+    when(materialization.nonWindowed().get(1,null, GKEY2)).thenReturn(Arrays.asList(ROW1).iterator());
+    when(materialization.nonWindowed().get(3,null, GKEY3)).thenReturn(Arrays.asList(ROW3).iterator());
+
```

Comment:
@AlanConfluent This test is supposed to test a range query "<= KEY3 " in a setting with multiple partitions. Originally I didn't have line 179 and  189 which I thought would get me the desired behavior, but somehow I ended up adding those lines. Now I'm not sure if the test is correctly testing the case I described or if there is a bug in the code how keys/partitions are used for range queries. What do you think?

Reply:
@agavra regarding your above comment about a mult-partition functional tests, is this what you had in mind?